babyagi.py

```diff
@@ -4,22 +4,39 @@
 import pinecone
 import time
 import sys
+import re
 from collections import deque
 from typing import Dict, List
 from dotenv import load_dotenv
-import os
 
 #Set Variables
 load_dotenv()
 
+# Log print output to txt file
+original_print = print
+def print(*tasks):
```

Comment:
Is it common in Python to clobber core functions like this? Seems like something that could (should?) live in its own function rather than always printing _everything_ to a file, no?

Reply:
Hello, thank you for reviewing my PR.
I chose this approach because it requires the least amount of code modification. I don't see many downsides of using this variant because we're using prints as the main interface to show data. But in the future, it could bring some complexity as we're not controlling what is logged to the file. I could rewrite this functionality to a decorator. I think it will maintain the low complexity of the code and we'll be more in control of what is being logged into the file.